tests/Fixer/Phpdoc/PhpdocParamOrderFixerTest.php

```diff
@@ -0,0 +1,473 @@
+     *
+     * @return bool
+     */
+    public function m($a, $b, $c, $d, $e) {}
+}
+EOT;
+        $this->doTest($expected, $input);
+    }
+
+    public function testOnlyParamsUndocumented()
+    {
+        $expected = <<<'EOT'
+<?php
+class C {
+    /**
+     * @param $a
+     * @param $b
+     * @param $c
+     * @param $d
+     */
+    public function m($a, $b, $c, $d, $e) {}
+}
+EOT;
+        $input = <<<'EOT'
+<?php
+class C {
+    /**
+     * @param $a
+     * @param $c
+     * @param $d
+     * @param $b
+     */
+    public function m($a, $b, $c, $d, $e) {}
+}
+EOT;
+        $this->doTest($expected, $input);
+    }
+
+    public function testOnlyParamsSuperfluousAnnotation()
+    {
+        $expected = <<<'EOT'
+<?php
+class C {
+    /**
+     * @param $a
+     * @param $b
+     * @param $c
+     * @param $superfluous
+     */
+    public function m($a, $b, $c) {}
+}
+EOT;
+        $input = <<<'EOT'
+<?php
+class C {
+    /**
+     * @param $a
+     * @param $superfluous
+     * @param $b
+     * @param $c
+     */
+    public function m($a, $b, $c) {}
+}
+EOT;
+        $this->doTest($expected, $input);
+    }
+
+    public function testOnlyParamsSuperfluousAnnotations()
+    {
+        $expected = <<<'EOT'
+<?php
+class C {
+    /**
+     * @param $a
+     * @param $b
+     * @param $c
+     * @param $superfluous2
+     * @param $superfluous1
+     * @param $superfluous3
+     */
+    public function m($a, $b, $c) {}
+}
+EOT;
+        $input = <<<'EOT'
+<?php
+class C {
+    /**
+     * @param $a
+     * @param $superfluous2
+     * @param $b
+     * @param $superfluous1
+     * @param $c
+     * @param $superfluous3
+     */
+    public function m($a, $b, $c) {}
+}
+EOT;
+        $this->doTest($expected, $input);
+    }
+
+    public function testParamsBlankLines()
+    {
+        $expected = <<<'EOT'
+<?php
+class C {
+    /**
+     * Some function
+     *
+     * @param $a
+     * @param $b
+     *
+     * @param $c
+     *
+     *
+     * @param $d
+     *
+     * @param $e
+     *
+     * @throws \Exception
+     *
+     * @return bool
+     */
+    public function m($a, $b, $c, $d, $e) {}
+}
+EOT;
+        $input = <<<'EOT'
+<?php
+class C {
+    /**
+     * Some function
+     *
+     * @param $b
+     * @param $e
+     *
+     * @param $c
+     *
+     *
+     * @param $a
+     *
+     * @param $d
+     *
+     * @throws \Exception
+     *
+     * @return bool
+     */
+    public function m($a, $b, $c, $d, $e) {}
+}
+EOT;
+        $this->doTest($expected, $input);
+    }
+
+    public function testPhp7()
+    {
+        $expected = <<<'EOT'
+<?php
+class C {
+    /**
+     * @param int $a
+     * @param $b
+     * @param bool $c
+     */
+    public function m(int $a, $b, bool $c) {}
+}
+EOT;
+        $input = <<<'EOT'
+<?php
+class C {
+    /**
+     * @param $b
+     * @param bool $c
+     * @param int $a
+     */
+    public function m(int $a, $b, bool $c) {}
+}
+EOT;
+        $this->doTest($expected, $input);
+    }
```

Comment:
Please add tests for comments between tokens:
```php
public/*1*/function/*2*/m/*3*/(/*4*/){}
```

Reply:
I've added one.